### PR TITLE
Add API rate limit check and wait

### PR DESCRIPTION
**Pull request by xtuchyna**
## This introduces a breaking change

- [ ] Yes
- [X] No

## This Pull Request implements
Now the API rate limit is checked after every analysis for each entity (after every Issue or PR)

### CODE REVIEW

File: srcopsmetrics/utils.py [hunk 1]
```diff
@@ -25,7 +25,53 @@
...
 from pathlib import Path
 
+from github import Github
+import time
+from datetime import datetime
+
 _LOGGER = logging.getLogger(__name__)
+API_RATE_MINIMAL_REMAINING = 20
+
+class knowledge:
```
Comment (pacospace) [inline srcopsmetrics/utils.py]: What about upper first letter for Classes? :)

Comment (xtuchyna) [reply srcopsmetrics/utils.py]: oh yeah! I don't know why I thought that context managers can be lower case :D

**Comment by xtuchyna** [comment]
dammit i really sometimes forget to format the code

**Comment by xtuchyna** [comment]
regate

**Pull request merged**